src/plots/cartesian/layout_attributes.js

```diff
@@ -478,6 +478,20 @@ module.exports = {
             'to the left/bottom of labels.'
         ].join(' ')
     },
+    ticklabelmode: {
+        valType: 'enumerated',
+        values: ['', 'period'],
+        dflt: '',
+        role: 'info',
+        editType: 'ticks',
+        description: [
+            'Determines where tick labels are drawn with respect to their',
+            'corresponding ticks and grid lines.',
+            'Only has an effect for axes of `type` *date*',
+            'When set to *period*, tick labels are drawn in the middle of the period',
+            'between ticks.'
```

Comment:
this is actually not quite the desired behaviour: if the ticks are spaced every 2 months and the labels are `Jan` and `Mar`, the `Jan` label should appear on Jan 15, *not* on Feb 1.

Reply:
Here's an example... https://codepen.io/nicolaskruchten/pen/ZEQRBWw?editors=0010 

In "instant" mode we see 4 "month" labels spaced two months apart. In "period" mode we probably should see only 3 labels, but they should remain Jan/Mar/May, and they should be positioned on Jan 15, Mar 15, May 15

Reply:
I'll add a clearer codepen in the original issue